src/build/functions/edge.ts

```diff
@@ -1,13 +1,43 @@
 import { cp, mkdir, readFile, rm, writeFile } from 'node:fs/promises'
-import { dirname, join } from 'node:path'
+import { dirname, join, relative } from 'node:path/posix'
 
 import type { Manifest, ManifestFunction } from '@netlify/edge-functions'
 import { glob } from 'fast-glob'
-import type { EdgeFunctionDefinition as NextDefinition } from 'next/dist/build/webpack/plugins/middleware-plugin.js'
+import type { FunctionsConfigManifest } from 'next-with-cache-handler-v2/dist/build/index.js'
+import type { EdgeFunctionDefinition as EdgeMiddlewareDefinition } from 'next-with-cache-handler-v2/dist/build/webpack/plugins/middleware-plugin.js'
 import { pathToRegexp } from 'path-to-regexp'
 
 import { EDGE_HANDLER_NAME, PluginContext } from '../plugin-context.js'
 
+type NodeMiddlewareDefinitionWithOptionalMatchers = FunctionsConfigManifest['functions'][0]
+type WithRequired<T, K extends keyof T> = T & { [P in K]-?: T[P] }
+type NodeMiddlewareDefinition = WithRequired<
+  NodeMiddlewareDefinitionWithOptionalMatchers,
+  'matchers'
+>
+
+function nodeMiddlewareDefinitionHasMatcher(
+  definition: NodeMiddlewareDefinitionWithOptionalMatchers,
+): definition is NodeMiddlewareDefinition {
+  return Array.isArray(definition.matchers)
+}
+
+type EdgeOrNodeMiddlewareDefinition = {
```

Comment:
Thoughts on keeping the two types separate for now and using something like `EdgeMiddlewareDefinition | NodeMiddlewareDefinition` downstream?